### ISSUE TITLE
[toggle-group] Missing Home / End keyboard navigation in ToggleGroup (ARIA spec)

### ISSUE THREAD

**Issue by jijiseong** [mui/base-ui#3970]
# Bug report

## Current behavior

Home and End keys do not move focus within the ToggleGroup.
Only arrow key navigation works.

## Expected behavior

According to the [WAI-ARIA Authoring Practices](https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/),
- Home should move focus to the first toggle
- End should move focus to the last toggle

## Reproducible example

1. goto base ui docs for toggle-group: https://base-ui.com/react/components/toggle-group#api-reference
2. click the first toggle item
3. press `End` key or `Home` key
4. not works

## Base UI version

 v1.1.0

## Which browser are you using?

Chrome/Safari etc.

## Which OS are you using?

Mac OS
